// @textile/grpc-powergate-client/dist/buildinfo/rpc/rpc_pb_service.js
// package: buildinfo.rpc
// file: buildinfo/rpc/rpc.proto

var buildinfo_rpc_rpc_pb = require("../../buildinfo/rpc/rpc_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var RPCService = (function () {
  function RPCService() {}
  RPCService.serviceName = "buildinfo.rpc.RPCService";
  return RPCService;
}());

RPCService.BuildInfo = {
  methodName: "BuildInfo",
  service: RPCService,
  requestStream: false,
  responseStream: false,
  requestType: buildinfo_rpc_rpc_pb.BuildInfoRequest,
  responseType: buildinfo_rpc_rpc_pb.BuildInfoResponse
};

exports.RPCService = RPCService;

function RPCServiceClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

RPCServiceClient.prototype.buildInfo = function buildInfo(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(RPCService.BuildInfo, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.RPCServiceClient = RPCServiceClient;
