// @textile/grpc-powergate-client/dist/buildinfo/rpc/rpc_pb.js
// source: buildinfo/rpc/rpc.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

goog.exportSymbol('proto.buildinfo.rpc.BuildInfoRequest', null, global);
goog.exportSymbol('proto.buildinfo.rpc.BuildInfoResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.buildinfo.rpc.BuildInfoRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.buildinfo.rpc.BuildInfoRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.buildinfo.rpc.BuildInfoRequest.displayName = 'proto.buildinfo.rpc.BuildInfoRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.buildinfo.rpc.BuildInfoResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.buildinfo.rpc.BuildInfoResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.buildinfo.rpc.BuildInfoResponse.displayName = 'proto.buildinfo.rpc.BuildInfoResponse';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.buildinfo.rpc.BuildInfoRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.buildinfo.rpc.BuildInfoRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.buildinfo.rpc.BuildInfoRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.buildinfo.rpc.BuildInfoRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.buildinfo.rpc.BuildInfoRequest}
 */
proto.buildinfo.rpc.BuildInfoRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.buildinfo.rpc.BuildInfoRequest;
  return proto.buildinfo.rpc.BuildInfoRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.buildinfo.rpc.BuildInfoRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.buildinfo.rpc.BuildInfoRequest}
 */
proto.buildinfo.rpc.BuildInfoRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.buildinfo.rpc.BuildInfoRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.buildinfo.rpc.BuildInfoRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.buildinfo.rpc.BuildInfoRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.buildinfo.rpc.BuildInfoRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.buildinfo.rpc.BuildInfoResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.buildinfo.rpc.BuildInfoResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.buildinfo.rpc.BuildInfoResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    gitCommit: jspb.Message.getFieldWithDefault(msg, 1, ""),
    gitBranch: jspb.Message.getFieldWithDefault(msg, 2, ""),
    gitState: jspb.Message.getFieldWithDefault(msg, 3, ""),
    gitSummary: jspb.Message.getFieldWithDefault(msg, 4, ""),
    buildDate: jspb.Message.getFieldWithDefault(msg, 5, ""),
    version: jspb.Message.getFieldWithDefault(msg, 6, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.buildinfo.rpc.BuildInfoResponse}
 */
proto.buildinfo.rpc.BuildInfoResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.buildinfo.rpc.BuildInfoResponse;
  return proto.buildinfo.rpc.BuildInfoResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.buildinfo.rpc.BuildInfoResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.buildinfo.rpc.BuildInfoResponse}
 */
proto.buildinfo.rpc.BuildInfoResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setGitCommit(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setGitBranch(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setGitState(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setGitSummary(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setBuildDate(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setVersion(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.buildinfo.rpc.BuildInfoResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.buildinfo.rpc.BuildInfoResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.buildinfo.rpc.BuildInfoResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getGitCommit();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getGitBranch();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getGitState();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getGitSummary();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getBuildDate();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getVersion();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
};


/**
 * optional string git_commit = 1;
 * @return {string}
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.getGitCommit = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.buildinfo.rpc.BuildInfoResponse} returns this
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.setGitCommit = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string git_branch = 2;
 * @return {string}
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.getGitBranch = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.buildinfo.rpc.BuildInfoResponse} returns this
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.setGitBranch = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string git_state = 3;
 * @return {string}
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.getGitState = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.buildinfo.rpc.BuildInfoResponse} returns this
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.setGitState = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string git_summary = 4;
 * @return {string}
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.getGitSummary = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.buildinfo.rpc.BuildInfoResponse} returns this
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.setGitSummary = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string build_date = 5;
 * @return {string}
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.getBuildDate = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.buildinfo.rpc.BuildInfoResponse} returns this
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.setBuildDate = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional string version = 6;
 * @return {string}
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.getVersion = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.buildinfo.rpc.BuildInfoResponse} returns this
 */
proto.buildinfo.rpc.BuildInfoResponse.prototype.setVersion = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


goog.object.extend(exports, proto.buildinfo.rpc);
