// @textile/grpc-powergate-client/dist/health/rpc/rpc_pb_service.js
// package: health.rpc
// file: health/rpc/rpc.proto

var health_rpc_rpc_pb = require("../../health/rpc/rpc_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var RPCService = (function () {
  function RPCService() {}
  RPCService.serviceName = "health.rpc.RPCService";
  return RPCService;
}());

RPCService.Check = {
  methodName: "Check",
  service: RPCService,
  requestStream: false,
  responseStream: false,
  requestType: health_rpc_rpc_pb.CheckRequest,
  responseType: health_rpc_rpc_pb.CheckResponse
};

exports.RPCService = RPCService;

function RPCServiceClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

RPCServiceClient.prototype.check = function check(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(RPCService.Check, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.RPCServiceClient = RPCServiceClient;
