// @textile/grpc-powergate-client/dist/index/ask/rpc/rpc_pb_service.js
// package: index.ask.rpc
// file: index/ask/rpc/rpc.proto

var index_ask_rpc_rpc_pb = require("../../../index/ask/rpc/rpc_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var RPCService = (function () {
  function RPCService() {}
  RPCService.serviceName = "index.ask.rpc.RPCService";
  return RPCService;
}());

RPCService.Get = {
  methodName: "Get",
  service: RPCService,
  requestStream: false,
  responseStream: false,
  requestType: index_ask_rpc_rpc_pb.GetRequest,
  responseType: index_ask_rpc_rpc_pb.GetResponse
};

RPCService.Query = {
  methodName: "Query",
  service: RPCService,
  requestStream: false,
  responseStream: false,
  requestType: index_ask_rpc_rpc_pb.QueryRequest,
  responseType: index_ask_rpc_rpc_pb.QueryResponse
};

exports.RPCService = RPCService;

function RPCServiceClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

RPCServiceClient.prototype.get = function get(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(RPCService.Get, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

RPCServiceClient.prototype.query = function query(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(RPCService.Query, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.RPCServiceClient = RPCServiceClient;
